static/js/chart/draw.ts

```diff
@@ -737,6 +822,60 @@ function drawLineChart(
   return !hasFilledInValues;
 }
 
+/**
+ * Adds tooltip element within a given container.
+ *
+ * @param containerId
+ */
+function addTooltip(containerId: string): void {
+  d3.select("#" + containerId)
+    .attr("style", "position: relative")
+    .append("div")
+    .attr("id", TOOLTIP_ID)
+    .attr("style", "position: absolute; display: none; z-index: 1");
+}
+
+/**
+ * Position and show the tooltip.
+ *
+ * @param text
+ * @param containerId
+ * @param datapointX
+ * @param datapointY
+ * @param relativeBoundary tooltip boundary relative to its container element
+ */
+function showTooltip(
+  text: string,
+  containerId: string,
+  datapointX: number,
+  datapointY: number,
+  relativeBoundary: Boundary
+): void {
+  const tooltipSelect = d3
+    .select("#" + containerId)
+    .select(`#${TOOLTIP_ID}`)
```

Comment:
should the css for #tooltip move to a common scss file as well?
also consider making it a more specific ID (e.g. draw-tooltip or chart-tooltip since we have a few other similarly id'ed elements in our codebase)

Reply:
done